src/traces/ohlc/helpers.js

```diff
@@ -0,0 +1,110 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+
+// This routine gets called during the trace supply-defaults step.
+//
+// This is a hacky way to make 'ohlc' and 'candlestick' trace types
+// go through the transform machinery.
+//
+// Note that, we must mutate user data (here traceIn) as opposed
+// to full data (here traceOut) as - at the moment - transform
+// defaults (which are called after trace defaults) start
+// from a clear transforms container. The mutations inflicted are
+// cleared in exports.clearEphemeralTransformOpts.
+exports.pushDummyTransformOpts = function(traceIn, traceOut) {
+    var transformOpts = {
+
+        // give dummy transform the same type as trace
+        type: traceOut.type,
+
+        // track ephemeral transforms in user data
+        _ephemeral: true
+    };
+
+    if(Array.isArray(traceIn.transforms)) {
+        traceIn.transforms.push(transformOpts);
+    }
+    else {
+        traceIn.transforms = [transformOpts];
+    }
+};
+
+// This routine gets called during the transform supply-defaults step
+// where it clears ephemeral transform opts in user data
+// and effectively put back user date in its pre-supplyDefaults state.
+exports.clearEphemeralTransformOpts = function(traceIn) {
```

Comment:
oh man... you know the transform implementation isn't :100: when you roll out a function like this. Fancy workarounds though :tada: